collectors/o365/o365_collector.js

```diff
@@ -168,7 +169,20 @@ class O365Collector extends PawsCollector {
 
             return callback(null, logs, newState, newState.poll_interval_sec);
         }).catch(err => {
-            if (err.message) {
+            // set errorCode to showcase client error on DDMetric
+            if (typeof err === 'object') {
+                err.errorCode = err.code ? err.code : (err.statusCode ? err.statusCode : err.status);
+            }
+            // Handle the MS management api maximum palyload issue
+            if (err.message && err.message.indexOf('Maximum payload size exceeded') !== -1) {
+                let min = moment(state.until).diff(state.since, 'minutes');
+                if (min > 1) {
+                    state.until = moment(state.since).add(min / 2, 'minutes').toISOString();
+                }
+                console.warn(`Reduce the time duration by ${min / 2} min due to Maximum payload issue`);
+                return callback(null, [], state, state.poll_interval_sec);
+            }
+            if (!err.code && err.message) {
```

Comment:
could you please also add a handler for expired content. I think we currently just fail the invocation instead of just acknowledging this error and proceeding with collection. We should probably handle this error in https://github.com/alertlogic/paws-collector/pull/193/files#diff-458b6045c8dfaadace71e6dbd57f1b5625d5c3f985af520c6b352b5111cff86cR150